engine/src/main/java/org/hibernate/search/v6poc/search/dsl/predicate/MatchPredicateFieldSetContext.java

```diff
@@ -14,6 +15,6 @@
  */
 public interface MatchPredicateFieldSetContext<N> extends MultiFieldPredicateFieldSetContext<MatchPredicateFieldSetContext<N>> {
 
-	N matching(Object value);
+	ExplicitEndContext<N> matching(Object value);
```

Comment:
So, in the spatial DSL interfaces (`SpatialWithinPredicateFieldSetContext` in particular), we return specific interfaces for each method: `SpatialWithinBoundingBoxPredicateContext`, `SpatialWithinCirclePredicateContext` and `SpatialWithinPolygonPredicateContext`. Here we just return the generic `ExplicitEndContext`. It feels a bit inconsistent.

In my opinion, what you did here is better. If we need to add more methods one day, we will be able to easily replace `ExplicitEndContext` in the signature of this method with a subinterface.

Don't you think we should do the same for the spatial DSL interfaces?

Note that, strangely, the "end" context interfaces in the spatial DSL do not extend `ExplicitEndContext`, even though they define an `end()` method.